lang/src/test/java/com/sap/psr/vulas/goals/BomGoalTest.java

```diff
@@ -127,7 +127,7 @@ public void testBomSkipSaveEmptyApp() throws GoalConfigurationException, GoalExe
 		verifyHttp(server).times(1, 
```

Comment:
In this test the application is uploaded (with a PUT) but from the test name i think it should have tested whether the upload is skipped. Right now it looks to me as if it's testing the same behavior than the next test "testBomSaveEmptyApp"

Reply:
True, the reason being that we ignore the configuration option `skipEmptyApps` if the application already exists in the backend. We basically have 2 options to fix the test:

  - We change `setupMockServices` such that it returns a 404 for OPTIONS .../app (in which case no PUT should be observed any more
  - We check that the application constructs and dependencies got deleted after the upload of the empty app

Reply:
Just implemented the first option